### DISCUSSION TITLE
<rh-card> Make the API more flexible

### DISCUSSION THREAD

**Discussion by markcaron** [RedHat-UX/red-hat-design-system#361] [Ideas]
IMO, we should make the API to card more flexible, as in not requiring a header or footer, so that it can be used as a building block with other components, like `<rh-stat>`, etc.

It would also open up the possibility for it to handle the need for an `Icon Panel`, which is basically a card with an icon at the top. I believe this would simplify things greatly.

**Comment by bennypowers** [comment]
currently card doesn't require `header` or `footer` and uses `SlotController` to hide those containers if the associated slotted content isn't present. (in the future we can maybe use `:has()` for this

**Comment by markcaron** [comment]
Do we extend the API of card to also cover "Sticky Card"?

<img width="302" alt="image" src="https://user-images.githubusercontent.com/1890950/172445101-052e5ec3-e09a-4578-8ceb-09c136283ec2.png">

XD mockup:
https://xd.adobe.com/view/d20a1c17-375c-4c7f-6827-5fe2f8c2d92f-a43b/screen/ab37d0fd-94c5-423a-ac5e-fe69951d492c/

**Reply by markcaron** [reply]
Also discussed in PFE:
https://github.com/patternfly/patternfly-elements/issues/1614

**Reply by bennypowers** [reply]
My initial impression is that the sticky card from that issue should either be:
a) its own specialized component (so as not to add excess unused code to the regular card), or
b) a modeless `<rh-dialog>`, perhaps even wrapped in a 3rd component
  ```html
  <rh-sticky>
    <rh-dialog open>
      <h4 slot=header>Card Title</h4>
      <h5 slot=description>Heading, xs</h5>
      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit,
         sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p> 
      <rh-cta><a href="/yadda/yadda">Call to action</a></rh-cta>
    </rh-dialog>
  </rh-sticky>
  ```
  From MDN:
  > The `<dialog>` [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML) element represents a dialog box or other interactive component, such as a dismissible alert, inspector, or subwindow.

  This matches up with "dismissable alert" nicely.
